pkg/logql/log/parser_hints.go

```diff
@@ -95,19 +91,25 @@ func (p *Hints) NoLabels() bool {
 }
 
 func (p *Hints) RecordExtracted(key string) {
-	for _, l := range p.requiredLabels {
-		if l == key {
-			p.extracted = append(p.extracted, key)
-			return
-		}
-	}
+	p.extracted = append(p.extracted, key)
 }
 
 func (p *Hints) AllRequiredExtracted() bool {
-	if len(p.requiredLabels) == 0 {
+	if len(p.requiredLabels) == 0 || len(p.extracted) < len(p.requiredLabels) {
 		return false
 	}
-	return len(p.extracted) == len(p.requiredLabels)
+
+	found := map[string]interface{}{}
```

Comment:
We probably want to say `found := make(map[string]struct{}, len(p.requiredLabels))` here. `struct{}` is smaller than `interface{}` and initializing with the max len we'll need ensures only one alloc